src/client/Microsoft.Identity.Client/PlatformsCommon/Shared/InMemoryPartitionedAppTokenCacheAccessor.cs

```diff
@@ -59,9 +65,25 @@ public void SaveAccessToken(MsalAccessTokenCacheItem item)
             string itemKey = item.CacheKey;
             string partitionKey = CacheKeyFactory.GetAppTokenCacheItemKey(item.ClientId, item.TenantId, item.KeyId, item.AdditionalCacheKeyComponents);
 
-            // if a conflict occurs, pick the latest value
-            AccessTokenCacheDictionary
-                .GetOrAdd(partitionKey, new ConcurrentDictionary<string, MsalAccessTokenCacheItem>())[itemKey] = item;
+            var partition = AccessTokenCacheDictionary.GetOrAdd(partitionKey, _ => new ConcurrentDictionary<string, MsalAccessTokenCacheItem>());
+            bool added = partition.TryAdd(itemKey, item);
+            if (added)
+            {
+                if (_tokenCacheAccessorOptions.UseSharedCache)
```

Comment:
You can extract this if else block in a method as 
```
private ref int GetEntryCountRef()
{
    return ref _tokenCacheAccessorOptions.UseSharedCache ? ref s_entryCount : ref _entryCount;
}
```
and then use it as 
```
if (added)
{
    Interlocked.Increment(ref GetEntryCountRef());
}
``` 
here and in the decrement logic

Reply:
Very cool, I had no idea